apps/query-ui/queryui/util.py

```diff
@@ -218,29 +220,36 @@ def format_value(value):
             return row
 
     def readdata(self):
-        """Read the trace data."""
-        directory = os.path.join(self.trace_dir, self.node_id)
-        docs = []
+        """Read the trace data for this node."""
+
+        if self.node_id not in self.result.execution:
+            return
+
+        directory = self.result.execution[self.node_id].trace_dir
+        if not directory:
+            return
 
         # We need to read all of the individual docs to ensure we get all of the parent
         # docs. With a very large number of docs, we are likely to blow out memory doing
         # this. Unfortunately there's no easy way to tell up-front that a given stage in
         # the pipeline has a mix of parent and child docs, unless we do two passes.
         # Just a heads up that with a larger number of docs, we may need to revisit this.
+        docs = []
         for filename in os.listdir(directory):
             f = os.path.join(directory, filename)
             if os.path.isfile(f):
+                self.num_files += 1
                 newdoc = self.readfile(f)
                 if newdoc:
                     docs.append(newdoc)
 
         # Only keep parent docs if there are child docs in the list.
         parent_docs = [x for x in docs if x.get("parent_id") is None]
```

Comment:
we might want to consider getting rid of this at some point now that we are using reconstructed documents in queries. Will be a way to enforce we're using the same model all through the stack